recipes-devtools/qemu/qemu-sel4-helpers/qemu-sel4

```diff
@@ -21,34 +17,84 @@ mkdir -p "$GUEST_SHARED"
 
 GUEST_GUI_ARGS="${GUEST_GUI_ARGS:--vga none -display none}"
 
-modprobe sel4_virt
+QEMU_APPEND_BOOTARGS=""
+QEMU_EXTRA_ARGS="${GUEST_GUI_ARGS}"
+
+# Decode args like 'uservm=1,0x48000000,0x10000000,0x60000000,0x10000000'
+# TODO: use device tree
+GUEST_MEMSZ=`while read -d' ' p; do
+    echo $p | grep '^uservm=1,' | cut -f3 -d, | xargs -Isz printf "%d" sz
+done < /proc/cmdline`
+
+if test -n "${GUEST_MEMSZ}"; then
+    # using seL4
+    GUEST_MEMSZ=`expr ${GUEST_MEMSZ} / 1048576`
+    ACCEL=sel4
+fi
+GUEST_MEMSZ=${GUEST_MEMSZ:-256}
+
+if test "x${ACCEL}" = "xsel4"; then
+    modprobe sel4_virt
+else
+    # using something else, let's default to KVM
+    QEMU_APPEND_BOOTARGS="${QEMU_APPEND_BOOTARGS} console=hvc0 root=/dev/vda2"
+    QEMU_EXTRA_ARGS="${QEMU_EXTRA_ARGS} -kernel /boot/Image -cpu host"
+    ACCEL=kvm
+fi
 
 ulimit -c unlimited
 
+ROOTFS_STORAGE=${ROOTFS_STORAGE:-tmpfs}
+case "${ROOTFS_STORAGE}" in
+    tmpfs)
+        cp ${GUEST_IMG} /tmp
+        GUEST_IMG=/tmp/`basename ${GUEST_IMG}`
+        echo "Using volatile rootfs cached in tmpfs" 1>&2
+        ;;
+    extra_ramdisk)
+        qemu-img create -f raw /tmp/ramdisk.raw 128M
+        mke2fs /tmp/ramdisk.raw
+        QEMU_EXTRA_ARGS="${QEMU_EXTRA_ARGS} -device virtio-blk-pci,drive=drive1,id=virtblk1"
+        QEMU_EXTRA_ARGS="${QEMU_EXTRA_ARGS} -drive format=raw,file=/tmp/ramdisk.raw,if=none,id=drive1"
+        echo "Using permanent storage for guest's rootfs" 1>&2
+        echo "Providing 128 MB /dev/vdb to guest" 1>&2
```

Comment:
I struggled with this a bit, because GUEST_IMG  elsewhere. So I think it would be worth mentioning that it is ram-backed:
```suggestion
        echo "Providing additional volatile 128 MB /dev/vdb to the guest" 1>&2
```